src/System.Numerics.Vectors/src/System/Numerics/Matrix4x4.cs

```diff
@@ -1755,8 +1760,29 @@ public static Matrix4x4 Transform(Matrix4x4 value, Quaternion rotation)
         /// </summary>
         /// <param name="matrix">The source matrix.</param>
         /// <returns>The transposed matrix.</returns>
-        public static Matrix4x4 Transpose(Matrix4x4 matrix)
+        public static unsafe Matrix4x4 Transpose(Matrix4x4 matrix)
         {
+#if HAS_INTRINSICS
+            if (Sse.IsSupported)
+            {
+                var row1 = Sse.LoadVector128(&matrix.M11);
```

Comment:
@CarolEidt, @fiigii:

The codegen for this is pretty sub-optimal:
```asm
       mov      rax, rdx
       vmovups  xmm0, xmmword ptr [rax]
       lea      rax, bword ptr [rdx+16]
       mov      r8, rax
       vmovups  xmm1, xmmword ptr [r8]
       lea      r8, bword ptr [rdx+32]
       mov      r9, r8
       vmovups  xmm2, xmmword ptr [r9]
       lea      r9, bword ptr [rdx+48]
       mov      r10, r9
       vmovups  xmm3, xmmword ptr [r10]
```

Reply:
Logged https://github.com/dotnet/coreclr/issues/19521 to track this one

Reply:
Nit: `var` => `Vector128<Matrix4x4>`, assuming that's what it is.  Same goes for other `var` usage... we only use `var` when the type is fully specified on the RHS.